Shared/Tests.Shared/RealmInstanceTest.cs

```diff
@@ -0,0 +1,44 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System.IO;
+using Realms;
+
+namespace IntegrationTests
+{
+    [Preserve(AllMembers = true)]
+    public abstract class RealmInstanceTest : RealmTest
+    {
+        protected RealmConfiguration _configuration = new RealmConfiguration(Path.GetTempFileName());
```

Comment:
Inheritors of this class can customize `_configuration` in their `SetUp` override and thus affect `_realm`. You should very rarely need to `Realm.GetInstance(_configuration)` and should just use `_realm` instead. They only case where it makes sense to not do that is when you need to open a new realm instance with the same configuration as `_realm`, so `Realm.GetInstance(_configuration)` is short for `Realm.GetInstance(_realm.Config)`.